partner_event/models/event_registration.py

```diff
@@ -0,0 +1,37 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import models, api
+
+
+class EventRegistration(models.Model):
+    _inherit = "event.registration"
+
+    def _prepare_partner(self, vals):
+        return {
+            'name': vals.get('name') or vals.get('email'),
+            'email': vals.get('email', False),
+            'phone': vals.get('phone', False),
+        }
+
+    @api.model
+    def create(self, vals):
+        if not vals.get('partner_id') and vals.get('email'):
```

Comment:
What if there is a name but not an email? Shouldn't it check existence / create the partner?